src/serde/reader.rs

```diff
@@ -46,10 +47,8 @@ pub enum DeserializeError {
     /// If decoding a message takes more than the provided size limit, this
     /// error is returned.
     SizeLimit,
-    SyntaxError(String),
-    EndOfStreamError,
-    UnknownFieldError,
-    MissingFieldError,
+    EndOfStream,
```

Comment:
We can very well special case `DeserializeError::Serde(de::value::Error::EndOfStreamError)`, no? I find it weird to have two different EndOfStream errors.